src/colmap/scene/database_cache.cc

```diff
@@ -209,4 +231,69 @@ const class Image* DatabaseCache::FindImageWithName(
   return nullptr;
 }
 
+bool DatabaseCache::setupPosePriors() {
+  LOG(INFO) << "Setting up prior positions...";
+
+  Timer timer;
+  timer.Start();
+
+  if (NumPosePriors() < 3) {
```

Comment:
I guess this is only valid for when only having position based priors.

Conceptually, it feels like the database cache should at most warn but not produce an error in this case. I would have expected the incremental mapper to error out in this case.

Reply:
You are completely right.  I have modified it so that it only returns false if there is no prior in the database.